web/src/lib/components/LoginModal.svelte

```diff
@@ -2,30 +2,35 @@
 
 <script lang="ts">
     import Login from "$lib/components/Login.svelte";
+    import { showLoginModal } from "$lib/stores";
 
-    let login : Login | null;
-    let isModalOpen = false;
+    let login: Login | null;
+
+    $: if ($showLoginModal?.opened ?? false) {
+        show($showLoginModal.callbackFunc)
+    }
 
     export function show(onLoginFunction) {
         if (typeof onLoginFunction === 'function') {
             onLogin = onLoginFunction;
         }
 
         login.startCheckingLogin();
-
-        isModalOpen = true;
     }
 
     export function hide() {
         login.stopCheckingLogin();
 
-        isModalOpen = false;
+        showLoginModal.set({
+            opened: false,
+            callbackFunc: null
+        });
     }
 
     export let onLogin: () => void = () => {};
 </script>
 
-<div class="modal" class:modal-open={isModalOpen}>
+<div class="modal" class:modal-open={$showLoginModal?.opened ?? false}>
```

Comment:
`opened` (that reflects the *real* state, not the "request to open") could be just a variable in this component. I don't think other components need to have access to that - all they should be able to do is to *request* the login modal to open itself.

Reply:
I had it that way with the `isModalOpen`, but I found out that I was doing

```
if (showLogonModal.opened) {
   isModalOpen = true
} else {
   isModalOpen = false
}
```

So why having another variable that is always set to the same value?